From 01556745f1a01a363c36fd2e5978dd4bec8fe97e Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Tue, 24 Sep 2024 19:44:54 +0200
Subject: [PATCH 1310/1695] dts: stm32: Fix clock source of STGEN to HSE_KER

The STGEN clock is simply derived from the external oscillator (HSE)
in most circumstances, no reason to use PLL4 here.

With this modification the STGEN can be preserved in Stop1 and LP-Stop1
mode if it is required for letency issue.

This patch is a preliminary step to support STGEN clock source in
RCC clock tree as any other peripheral clocks.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 64471ac3027fa2931b4dc47640c94f2193d7381b dts: stm32: Fix clock source of STGEN to HSE_KER]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I0ef087f0a451e4e43acc316bc69a015218bb9630
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/406622
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rcc.dtsi  | 2 +-
 core/arch/arm/dts/stm32mp235f-dk-ca35tdcid-rcc.dtsi  | 2 +-
 core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rcc.dtsi  | 2 +-
 core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rcc.dtsi | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rcc.dtsi b/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rcc.dtsi
index 8f09c5cb3..77a9bc7cb 100644
--- a/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rcc.dtsi
+++ b/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rcc.dtsi
@@ -72,7 +72,7 @@
 		FLEXGEN_CFG(29, XBAR_SRC_PLL5, 0, 1)
 		FLEXGEN_CFG(30, XBAR_SRC_HSE_KER, 0, 1)
 		FLEXGEN_CFG(31, XBAR_SRC_PLL5, 0, 19)
-		FLEXGEN_CFG(33, XBAR_SRC_PLL4, 0, 23)
+		FLEXGEN_CFG(33, XBAR_SRC_HSE_KER, 0, 0)
 		FLEXGEN_CFG(36, XBAR_SRC_PLL4, 0, 11)
 		FLEXGEN_CFG(37, XBAR_SRC_PLL5, 0, 3)
 		FLEXGEN_CFG(38, XBAR_SRC_PLL4, 0, 11)
diff --git a/core/arch/arm/dts/stm32mp235f-dk-ca35tdcid-rcc.dtsi b/core/arch/arm/dts/stm32mp235f-dk-ca35tdcid-rcc.dtsi
index eeeeb314e..d953c588e 100644
--- a/core/arch/arm/dts/stm32mp235f-dk-ca35tdcid-rcc.dtsi
+++ b/core/arch/arm/dts/stm32mp235f-dk-ca35tdcid-rcc.dtsi
@@ -74,7 +74,7 @@
 		FLEXGEN_CFG(30, XBAR_SRC_HSE_KER, 0, 1)
 		FLEXGEN_CFG(31, XBAR_SRC_PLL5, 0, 19)
 		FLEXGEN_CFG(32, XBAR_SRC_PLL5, 0, 19)
-		FLEXGEN_CFG(33, XBAR_SRC_PLL4, 0, 23)
+		FLEXGEN_CFG(33, XBAR_SRC_HSE_KER, 0, 0)
 		FLEXGEN_CFG(34, XBAR_SRC_PLL4, 0, 59)
 		FLEXGEN_CFG(35, XBAR_SRC_HSI_KER, 0, 3)
 		FLEXGEN_CFG(36, XBAR_SRC_PLL5, 0, 3)
diff --git a/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rcc.dtsi b/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rcc.dtsi
index e935534b4..c246ad101 100644
--- a/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rcc.dtsi
+++ b/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rcc.dtsi
@@ -67,7 +67,7 @@
 		FLEXGEN_CFG(30, XBAR_SRC_HSE_KER, 0, 1)
 		FLEXGEN_CFG(31, XBAR_SRC_PLL5, 0, 19)
 		FLEXGEN_CFG(32, XBAR_SRC_PLL5, 0, 19)
-		FLEXGEN_CFG(33, XBAR_SRC_PLL4, 0, 23)
+		FLEXGEN_CFG(33, XBAR_SRC_HSE_KER, 0, 0)
 		FLEXGEN_CFG(34, XBAR_SRC_PLL4, 0, 59)
 		FLEXGEN_CFG(35, XBAR_SRC_HSI_KER, 0, 3)
 		FLEXGEN_CFG(36, XBAR_SRC_PLL5, 0, 3)
diff --git a/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rcc.dtsi b/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rcc.dtsi
index 8c9314fed..71f76e071 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rcc.dtsi
+++ b/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rcc.dtsi
@@ -67,7 +67,7 @@
 		FLEXGEN_CFG(30, XBAR_SRC_HSE_KER, 0, 1)
 		FLEXGEN_CFG(31, XBAR_SRC_PLL5, 0, 19)
 		FLEXGEN_CFG(32, XBAR_SRC_PLL5, 0, 19)
-		FLEXGEN_CFG(33, XBAR_SRC_PLL4, 0, 23)
+		FLEXGEN_CFG(33, XBAR_SRC_HSE_KER, 0, 0)
 		FLEXGEN_CFG(34, XBAR_SRC_PLL4, 0, 59)
 		FLEXGEN_CFG(35, XBAR_SRC_HSI_KER, 0, 3)
 		FLEXGEN_CFG(36, XBAR_SRC_PLL5, 0, 3)
-- 
2.47.2

